src/tuesday/core/instrument_models/noise.py

```diff
@@ -0,0 +1,197 @@
+"""A module to add thermal noise to lightcones."""
+
+import warnings
+from collections.abc import Callable
+
+import astropy.units as un
+import numpy as np
+from astropy.constants import c
+from astropy.cosmology import Planck18
+from astropy.cosmology.units import littleh
+from py21cmsense import Observation
+from py21cmsense.conversions import dk_du, f2z
+from scipy.signal import windows
+
+
+def grid_baselines(uvws, freq, boxlength: un.Quantity, lc_shape, weights):
+    r"""Grid baselines in uv space.
+
+    Parameters
+    ----------
+    uvws : np.ndarray
+        Baselines in uv space with shape (N bls, N time offsets, 3).
+    freq : un.Quantity
+        Frequency at which the baselines are projected.
+    boxlength : un.Quantity
+        Length of the box in which the lightcone is defined.
+    lc_shape : tuple
+        Shape of the lightcone (Nx, Ny, Nz).
+        We assume that Nx = Ny.
+    weights : np.ndarray
+        Weights for each baseline group with shape (N bls).
+
+    Returns
+    -------
+    uvsum : np.ndarray
+        2D histogram of uv counts for one day
+        of observation with shape (Nu=Nx, Nv=Nx).
+
+    """
+    dx = float(boxlength.value) / float(lc_shape[0])
+    ugrid_edges = (
+        np.fft.fftshift(np.fft.fftfreq(lc_shape[0], d=dx))
+        * 2
+        * np.pi
+        / Planck18.h
+        * littleh
+        / un.Mpc
```

Comment:
Then here, divide by the units of `boxlength` instead of assuming they are Mpc/h